# p-Coumaric acid production in *Synechocystis* sp. PCC6803

#### Students

Victor Buñuel Sorribas (s212736)





## 1. Introduction

### 1.1. Literature review of the compound

### 1.2. Literature review of the cell factory

Cyanobacteria are photosynthetic unicellular prokaryotic organisms that have been present on earth for the last 2.5 billion years (Brocks, 1999). Cyanobacteria are considered the firs oxygenic photosynthetic microorganisms contributing to oxygen production on Earth (Rasmussen, 2008). They can grow as individual cells, filamentus or colonies, creating big blooms that are visible (Catherine, 2013). 

Synechocystis sp. strain PCC6803 is one of the most studied cyanobacteria species, and it was first isolated from a lake in 1968 (Yu, 2013). Its genome size is 3.6 Mb, and it can be easily engineered. The genome of Synechocystis PCC6803 was sequenced by Kaneko et al. (1996), and it was the first genomic sequence obtained from a photosynthetic autotroph microorganism. Synechocystis PCC6803 can either use the Calvin cycle to fix CO2 as a photoautotrophic or use glycolysis to metabolize glucose as heterotrophic. It can also show a photoheterotrophic metabolism. However, cellular growth is around 8-fold higher in photoautotrophic (7-10 hours) than in heterotrophic metabolism (84 hours) (Yu, 2013).  Genetic modification studies shown that Synechocystis PCC6803 has a very plastic metabolism, being able to adapt from photoautotrophic to heterotrophic growing conditions (Vermaas, 1996). 

Cyanobacteria are very good potential cell factory platform because of its fast growth rate, the sustainable process when fixing CO2 and the easily engineered genome. There are also some challenges that must be solved in the future, such as low product titer, long biosynthesis process or production and scale-up costs (Yu, 2013). Cyanobacteria have been used for production of several biofuels such as ethanol, butanol or isobutanol (Nozzi, 2013); pharmaceutical bioactive compounds (Vijayakumar, 2015) and other chemical compound as polyketides, terpenes, alkaloids (Gademann, 2011). 

Different genome-scale metabolic models have addressed Synechocystis PCC6803. First genome-scale network was characterized by Shastri & Morgan (2005) using a Flux Balance Analysis (FBA). In order to improve the multiple solutions obtained with this first stoichiometric-based FBA model, other methods were used. Flux Coupling Finder (FCF) was used, in combination with transcriptomics, to elucidate the metabolic modifications that lead to change from autotrophic to heterotrophic metabolism as a respond to availability of light (Montagud, 2011). Recently, FBA has been used to build a new genomic-scale model with a better characterization of the photo-respiratory metabolism. Moreover, this model used Dynamic Flux Balance Analysis (DFBA) to consider the metabolic changes due to light-dark cycle (Joshi, 2017).  


## 2. Problem definition

## 3. Selection and assessment of existing GSM

In [640]:
import pytest

In [641]:
import hashlib
import numpy as np
from cobra.io import read_sbml_model
from cobra.util import create_stoichiometric_matrix
from cobra import Reaction, Metabolite

In [642]:
model = read_sbml_model('iSynCJ816knock.xml')

No objective coefficients in model. Unclear what should be optimized


In [643]:
#adding the metabolites (from KEGG, according to the article)


In [644]:
trans_Cinnamate = Metabolite(id = 'TRCI', compartment='c', name='trans-Cinnamate', formula='C9H8O2')

In [645]:
p_Coumaric_acid = Metabolite(id = 'PCOU', compartment='c', name='p-Coumaric acid', formula='C9H8O3')

In [646]:
#adding the reactions

In [647]:
phenylalanine_ammonia_lyase = Reaction('PAL')

In [648]:
trans_cinnamate_4_monooxygenase = Reaction('CYP73A')

In [649]:
#adding the metabolites into the reactions

In [650]:
phenylalanine_ammonia_lyase.add_metabolites(({model.metabolites.phe__L_c : -1,  
                              trans_Cinnamate: 1, 
                              model.metabolites.nh3_c: 1, 
                            }))

In [651]:
trans_cinnamate_4_monooxygenase.add_metabolites(({model.metabolites.o2_c : -1,
                                                  trans_Cinnamate: -1,
                                                  model.metabolites.nadph_c : -1,
                                                  p_Coumaric_acid: 1,
                                                  model.metabolites.h2o_c: 1,
                                                  model.metabolites.nadph_c : 1,
                                                 }))

In [652]:
#Adding the reactions to the model:
model.add_reactions([phenylalanine_ammonia_lyase])
model.add_reactions([trans_cinnamate_4_monooxygenase])

In [653]:
model.add_boundary(model.metabolites.TRCI, type='demand')
model.add_boundary(model.metabolites.PCOU, type='demand')

<Reaction DM_PCOU at 0x2a9f0037a00>




### 3.¿? Study of important fluxes

***p-coumaric production analysis***

p-coumaric range of production is expressed as mmol p-coumaric/g(DW)h. Using experimental data, we can establish the bounds for CYP73A reaction. Gao et al. (2021) performed different Synechocystis sp. PCC 6803 tests, where the cyanobacteria were cultured during 7 days (168 hours) reaching a OD730 of 1,4 and then the concentration of p-coumaric acid measured, with results in the range 130-200 mg/L. The DW concentration of bacteria can be calculated as 148 mg/L/OD730 (Du, 2016). Calculations to obtain the final values and results can be seen below.

In [654]:
DW_factor = 0.148 #g/L/OD730
OD730 = 1.24 
DW_concentration = DW_factor * OD730
print('DW_concentration =', DW_concentration, 'g(DW)/L')
Time = 168 #hours
MW = 164 #g/mol
Coumaric_max_g = 0.200 #g/l
Coumaric_min_g = 0.130 #g/L
Coumaric_max_mmol = Coumaric_max_g / MW * 1000 
print('Coumaric_max_mmol/L', Coumaric_max_mmol, 'mmol/L')
Coumaric_min_mmol = Coumaric_min_g / MW * 1000 
print('Coumaric_min_mmol/L', Coumaric_min_mmol, 'mmol/L')
Productivity_max=Coumaric_max_mmol/DW_concentration/Time
Productivity_min=Coumaric_min_mmol/DW_concentration/Time
print('Productivity_max =', Productivity_max, 'mmol/g(DW)/h') 
print('Productivity_min =', Productivity_min, 'mmol/g(DW)/h') 

DW_concentration = 0.18352 g(DW)/L
Coumaric_max_mmol/L 1.2195121951219512 mmol/L
Coumaric_min_mmol/L 0.7926829268292683 mmol/L
Productivity_max = 0.03955427834263397 mmol/g(DW)/h
Productivity_min = 0.025710280922712082 mmol/g(DW)/h


In [655]:
with model as model:
    model.objective=model.reactions.CYP73A
    maximum_growth_hetero = model.optimize().objective_value #flux
    print('Maximum p-coumaric =', maximum_growth_hetero, 'mmol/g(DW)/h')

Maximum p-coumaric = 3.484935297244293e-05 mmol/g(DW)/h


As we can see, the difference between the experimental p-coumaric production and the one predicted by the model is very high. Thus,from this comparation can be concluded that the model needs a lot of manual curing and optimization to be able to predict the p-coumaric production and be realistic. However, in the next analysis we will use the predicted value as reference to see if p-coumaric production can be improved by genome editing techniques. 

***Photon exchange analysis***

In [656]:
with model as model:
    model.objective=model.reactions.EX_photon_e
    maximum_growth_hetero = model.optimize().objective_value #flux
    print('Maximum photon exchange =', maximum_growth_hetero, 'mmol/g(DW)/h')
    print('Photon exchange bounds =', model.reactions.EX_photon_e.bounds)

Maximum photon exchange = 0.0 mmol/g(DW)/h
Photon exchange bounds = (0.0, 0.0)


Photons are essential for any photosynthetic microorganism. The model doesn't consider photon exchange, as the bounds are set to 0. A bioenergetic assesment carried out by Touloupakis et al. (2015) showed that photon exchange flux in *Synechocystis* is ranged up to 95 mmol/g(DW)/h, being optimal at 50 mmol/g(DW)/h. Bounds are set as [-50, 0] for autotrophic conditions. 

***Glucose uptake analysis***

In [680]:
with model as model:
    model.objective=model.reactions.BIOMASS_Ec_SynAuto_1
    maximum_growth_hetero = model.optimize().objective_value #flux
    glucose=model.reactions.EX_glc__D_e.flux
    print('Maximum glucose exchange =', glucose, 'mmol/g(DW)/h')
    print('Photon exchange bounds =', model.reactions.EX_glc__D_e.bounds)

Maximum glucose exchange = -0.85 mmol/g(DW)/h
Photon exchange bounds = (-0.85, 999999.0)


In [693]:
with model as model:
    model.objective=model.reactions.BIOMASS_Ec_SynMixo_1
    maximum_growth_hetero = model.optimize().objective_value #flux
    glucose=model.reactions.EX_glc__D_e.flux
    print('Maximum glucose exchange =', glucose, 'mmol/g(DW)/h')
    print('Photon exchange bounds =', model.reactions.EX_glc__D_e.bounds)

Maximum glucose exchange = -0.85 mmol/g(DW)/h
Photon exchange bounds = (-0.85, 999999.0)


Glucose uptake is done under heterotrophic and mixotrophic mode. Our model shows a glucose uptake rate of 0.85 mmol/g(DW)/h for both heterotrophic and mixotrophic modes. Metabolic flux analysis carried out by Yang (2002) shows that 0.85 and 0.38 are the rates for heterotrophic and mixotrophic modes, respectively. 

***CO2 fixation (RuBisCO) analysis***

In [705]:
with model as model:
    model.objective=model.reactions.RBPC_1
    co2fix = model.optimize().objective_value #flux
    print('Maximum CO2 fixation rate =', co2fix, 'mmol/g(DW)/h')
    print('CO2 exchange bounds =', model.reactions.RBPC_1.bounds)


Maximum CO2 fixation rate = 71435.92738095242 mmol/g(DW)/h
CO2 exchange bounds = (0.0, 999999.0)


CO2 fixation is the main process that photosynthetic microorganisms use to obtain carbon. CO2 is incorporated to ribulose-1,5-biphosphate, in a reaction catalyzed by RuBisCO (RBPC_1). Thus, we use flux trhough RuBisCO as a way to measure CO2 fixation. The model predicts a optimized value that is too high, so bounds must be changed to perform model analysis. A characterization of *Synechocystis* sp- PC6803 carried out by Zavřel et al. (2017), showed CO2 fixation rate range to be 2.97 - 3.54 mmol(CO2)/g(DW)/h. Bounds are set as [2.97, 3.54] for autotrophic and mixotrophic conditions.


### 3.2. Autotrophic, heterotrophic and mixotrophic comparation

It has already been explained in the literature review of the cell factory section how *Synechocystis* sp. PCC6803 can switch between autotrophic, heterotrophic and mixotrophic metabolism. Our model considers this three metabolisms, and has a specific biomass reaction for each of them. By setting each biomass reaction as objective and optimizing the flux through it the model will measure the rest ofthe metabolic flux as autotrophic, mixotrophic or heterotrophic behaviour.
In addition, some other paramenters are modified to optimize the metabolism mode simulation. The model doesn't consider flux through CO2 fixation and photon exchange, so specific RBPC_1 and EX_photon_e bounds are established using literature data.

In [685]:
#Autotrophic
with model as model:
    model.reactions.EX_photon_e.bounds = -50, 0
    model.reactions.RBPC_1.bounds = 2.97, 3.54
    model.objective=model.reactions.BIOMASS_Ec_SynAuto_1
    maximum_growth_hetero = model.optimize().objective_value #flux
    print('Maximum growth =', maximum_growth_hetero, 'h-1')
    p_coumaric_production = model.reactions.CYP73A.flux
    print('Maximum productivity =', p_coumaric_production, 'mmol p-coumaric/gDW*h')
    maximum_yield_co2 = p_coumaric_production / (1*(model.reactions.RBPC_1.flux)) #yield calculations
    print('Maximum theoretical yield on co2 =', maximum_yield_co2, 'mmol-p_coumaric/mmol-co2')
    cmolmaximum_yield_co2 = p_coumaric_production*9 / (1*(model.reactions.RBPC_1.flux)) #yield calculations
    print('C-mmol Maximum theoretical yield on co2 =', cmolmaximum_yield_co2, 'c-mmol-p_coumaric/c-mmol-co2')

Maximum growth = 0.17433033139529122 h-1
Maximum productivity = 7.78681291198624e-05 mmol p-coumaric/gDW*h
Maximum theoretical yield on co2 = 2.6218225292882964e-05 mmol-p_coumaric/mmol-co2
C-mmol Maximum theoretical yield on co2 = 0.00023596402763594666 c-mmol-p_coumaric/c-mmol-co2


In [686]:
#Mixotrophic
with model as model:
    model.reactions.RBPC_1.bounds = 2.97, 3.54
    model.objective=model.reactions.BIOMASS_Ec_SynMixo_1
    maximum_growth_mixo = model.optimize().objective_value #flux
    print('Maximum growth =', maximum_growth_mixo, 'h-1')
    p_coumaric_production = model.reactions.CYP73A.flux
    print('Maximum productivity =', p_coumaric_production, 'mmol p-coumaric/gDW*h')
    maximum_yield_co2 = p_coumaric_production / (1*(model.reactions.RBPC_1.flux)) #yield calculations
    print('Maximum theoretical yield on co2 =', maximum_yield_co2, 'mmol-p_coumaric/mmol-co2')
    maximum_yield_glucose = p_coumaric_production / (-1*(model.reactions.EX_glc__D_e.flux)) #yield calculations
    print('Maximum theoretical yield on glucose =', maximum_yield_glucose, 'mmol-p-coumaric/mmol-glucose')
    cmolmaximum_yield_co2 = p_coumaric_production*9 / (1*(model.reactions.RBPC_1.flux)) #yield calculations
    print('C-mmol Maximum theoretical yield on co2 =', cmolmaximum_yield_co2, 'c-mmol-p_coumaric/c-mmol-co2')
    cmolmaximum_yield_glucose = p_coumaric_production*9 / (-1*(model.reactions.EX_glc__D_e.flux)*6) #yield calculations
    print('C-mmol Maximum theoretical yield on glucose =', cmolmaximum_yield_glucose, 'c-mmol-p-coumaric/c-mmol-glucose')

Maximum growth = 0.17777590204169774 h-1
Maximum productivity = 7.940716216598559e-05 mmol p-coumaric/gDW*h
Maximum theoretical yield on co2 = 2.6141374371291162e-05 mmol-p_coumaric/mmol-co2
Maximum theoretical yield on glucose = 9.342019078351247e-05 mmol-p-coumaric/mmol-glucose
C-mmol Maximum theoretical yield on co2 = 0.00023527236934162046 c-mmol-p_coumaric/c-mmol-co2
C-mmol Maximum theoretical yield on glucose = 0.0001401302861752687 c-mmol-p-coumaric/c-mmol-glucose


In [660]:
#Heterotrophic
with model as model:
    model.reactions.RBPC_1.bounds = 0, 0
    model.objective=model.reactions.BIOMASS_Ec_SynHetero_1
    maximum_growth_hetero = model.optimize().objective_value #flux
    print('Maximum growth =', maximum_growth_hetero, 'h-1')
    p_coumaric_production = model.reactions.CYP73A.flux
    print('Maximum productivity =', p_coumaric_production, 'mmol p-coumaric/gDW*h')
    maximum_yield_glucose = p_coumaric_production / (-1*(model.reactions.EX_glc__D_e.flux)) #yield calculations
    print('Maximum theoretical yield on glucose =', maximum_yield_glucose, 'mmol-p-coumaric/mmol-glucose')
    cmolmaximum_yield_glucose = p_coumaric_production*9 / (-1*(model.reactions.EX_glc__D_e.flux)*6) #yield calculations
    print('C-mmol Maximum theoretical yield on glucose =', cmolmaximum_yield_glucose, 'c-mmol-p-coumaric/c-mmol-glucose')

Maximum growth = 0.07802010063372698 h-1
Maximum productivity = 3.484923835037974e-05 mmol p-coumaric/gDW*h
Maximum theoretical yield on glucose = 4.0999103941623224e-05 mmol-p-coumaric/mmol-glucose
C-mmol Maximum theoretical yield on glucose = 6.149865591243484e-05 c-mmol-p-coumaric/c-mmol-glucose


##### Results

|  | Max. growth (h-1) | Max. productivity (mmol/g(DW)/h)| Max. yield on glucose | Max yield on CO2 | C-mmol max yield on glucose | C-mmol max yield on CO2 |
| :-: | :-: | :-: | :-: | :-: | :-: | :-: |
| Autotrophic | 0.17 | 7.78E-5| --- | 2.62E-5 | --- | 2.35E-4 |
| Mixootrophic | 0.078 |3.48E-5 | 4.09E-5 | 1.17E-5 | 6.14E-5 | 1.05E-4 |
| Heterotrophic | 0.078 | 3.48E-5 | 4.09E-5 | --- |  6.14E-5 | --- |


Results are very similar in the three conditions. Yang et al. (2002) reported a maximum heterotrophic and mixotrophic growth rate of 0.076 and 0.059 h-1, respectively. For the autotrophic growth rate, van Alphen et al. (2018) reported 0.16 h-1 under optimized light conditions, although standard conditions show 0.09 h-1. Thus, growth rates predicted by our models are in the range of literature data and are valid. 

p-coumaric production is equal to the result obtained for the optimization of CYP73A flux. In this sense, our model predicts a low p-coumaric production but this production remains constant when the biomass growth is optimized. Thus, the product can be biosynthesized under optimal growth conditions. 

Finally, C-mmol yields on glucose and CO2 show that CO2 asimilation leads to a higher c-mmol production of p-coumaric acid. Autotrophic mode is feasible in terms of sustainability, as more carbon from the external source end up in p-coumaric acid molecule than in heterotrophic or mixotrophic modes. 

## 4. Computer-Aided Cell Factory Engineering

We will focus the cell-factory engineering on metabolic engineering strategies to increase p-coumaric production. First, the main metabolic pathways leading to p-coumaric synthesis were analysed, using KEGG to browse the following specific *Synechocystis* sp. PCC6803 pathways: Aromatic aminoacids metabolism (https://www.genome.jp/pathway/syn00400) and phenylalanine metabolism (https://www.genome.jp/pathway/syn00360). Two strategies will be addressed: phenylalanine flux optimization and addition of new biosynthetic pathways. 

### 4.1. p-coumaric flux optimization: PPOR and PPND knockout

Two different knockout strategies were tested to optimize p-coumaric flux. Knocout targets were selected based on KEGG pathways metabolic analysis and similar experiments from literature. These knockouts would theoretically reduce competition and would redirect more carbon to p-coumaric production pathway. 

The phenylpyruvate oxidoreductase (PPOR) is an enzyme catalyzing the conversion of phenylpyruvate into 2-hydroxyphenylacetate. Phenylpyruvate is the precursor of phenylalanine, so in theory PPOR knockout would increase the flux through phenylalanine biosynthesis. 

The prephenate dehydrogenase (PPND) catalyzes the conversion of prephenate into 3-4-Hydroxyphenylpiruvate, which will be converted into tyrosine by tyrosine transaminase (TYRTA) in the next reaction. Knockout of tyrosine biosynthetic enzyme genes has shown to increase flux through p-coumaric production without disturbing *Synechocystis* metabolism (Gao, 2021). Hence, we decided to knockout PPND. 

In [689]:
#Autotrophic
with model as model:
    model.reactions.PPND.bounds = 0, 0
    model.reactions.PPOR.bounds = 0, 0
    print("PPND's bounds inside the with statement:")
    print(model.reactions.PPND.lower_bound, model.reactions.PPND.bounds)
    print("PPOR's bounds inside the with statement:")
    print(model.reactions.PPOR.lower_bound, model.reactions.PPOR.bounds)
    model.reactions.EX_photon_e.bounds = -50, 0
    model.reactions.RBPC_1.bounds = 2.97, 3.54
    model.objective=model.reactions.BIOMASS_Ec_SynAuto_1
    maximum_growth_hetero = model.optimize().objective_value #flux
    print('Maximum growth =', maximum_growth_hetero, 'h-1')
    p_coumaric_production = model.reactions.CYP73A.flux
    print('Maximum productivity =', p_coumaric_production, 'mmol p-coumaric/gDW*h')
    maximum_yield_co2 = p_coumaric_production / (1*(model.reactions.RBPC_1.flux)) #yield calculations
    print('Maximum theoretical yield on co2 =', maximum_yield_co2, 'mmol-p_coumaric/mmol-co2')

PPND's bounds inside the with statement:
0 (0, 0)
PPOR's bounds inside the with statement:
0 (0, 0)
Maximum growth = 3.432938797288907e-17 h-1
Maximum productivity = 4.519548762969066e-15 mmol p-coumaric/gDW*h
Maximum theoretical yield on co2 = 1.5217335902252745e-15 mmol-p_coumaric/mmol-co2


In [662]:
#Mixotrophic
with model as model:
    model.reactions.PPND.bounds = 0, 0
    model.reactions.PPOR.bounds = 0, 0
    print("PPND's bounds inside the with statement:")
    print(model.reactions.PPND.lower_bound, model.reactions.PPND.bounds)
    print("PPOR's bounds inside the with statement:")
    print(model.reactions.PPOR.lower_bound, model.reactions.PPOR.bounds)
    model.reactions.EX_photon_e.bounds = 0, 95
    model.reactions.RBPC_1.bounds = 2.97, 3.54
    model.objective=model.reactions.BIOMASS_Ec_SynMixo_1
    maximum_growth_mixo = model.optimize().objective_value #flux
    print('Maximum growth =', maximum_growth_mixo, 'h-1')
    p_coumaric_production = model.reactions.CYP73A.flux
    print('Maximum productivity =', p_coumaric_production, 'mmol p-coumaric/gDW*h')
    maximum_yield_co2 = p_coumaric_production / (1*(model.reactions.RBPC_1.flux)) #yield calculations
    print('Maximum theoretical yield on co2 =', maximum_yield_co2, 'mmol-p_coumaric/mmol-co2')
    maximum_yield_glucose = p_coumaric_production / (-1*(model.reactions.EX_glc__D_e.flux)) #yield calculations
    print('Maximum theoretical yield on glucose =', maximum_yield_glucose, 'mmol-p-coumaric/mmol-glucose')

PPND's bounds inside the with statement:
0 (0, 0)
PPOR's bounds inside the with statement:
0 (0, 0)
Maximum growth = 1.6368899029428874e-16 h-1
Maximum productivity = -1.3670761137979829e-15 mmol p-coumaric/gDW*h
Maximum theoretical yield on co2 = -4.602949878107686e-16 mmol-p_coumaric/mmol-co2
Maximum theoretical yield on glucose = -1.6083248397623328e-15 mmol-p-coumaric/mmol-glucose


In [663]:
#Heterotrophic
with model as model:
    model.reactions.PPND.bounds = 0, 0
    model.reactions.PPOR.bounds = 0, 0
    print("PPND's bounds inside the with statement:")
    print(model.reactions.PPND.lower_bound, model.reactions.PPND.bounds)
    print("PPOR's bounds inside the with statement:")
    print(model.reactions.PPOR.lower_bound, model.reactions.PPOR.bounds)
    model.reactions.RBPC_1.bounds = 0, 0
    model.objective=model.reactions.BIOMASS_Ec_SynHetero_1
    maximum_growth_hetero = model.optimize().objective_value #flux
    print('Maximum growth =', maximum_growth_hetero, 'h-1')
    p_coumaric_production = model.reactions.CYP73A.flux
    print('Maximum productivity =', p_coumaric_production, 'mmol p-coumaric/gDW*h')
    maximum_yield_glucose = p_coumaric_production / (-1*(model.reactions.EX_glc__D_e.flux)) #yield calculations
    print('Maximum theoretical yield on glucose =', maximum_yield_glucose, 'mmol-p-coumaric/mmol-glucose')

PPND's bounds inside the with statement:
0 (0, 0)
PPOR's bounds inside the with statement:
0 (0, 0)
Maximum growth = 0.0 h-1
Maximum productivity = 0.0 mmol p-coumaric/gDW*h
Maximum theoretical yield on glucose = 0.0 mmol-p-coumaric/mmol-glucose


##### Results

|  | Max. growth (h-1) | WT Max. productivity (mmol/g(DW)/h) | XAL Max. productivity (mmol/g(DW)/h)| Max. yield on glucose | Max yield on CO2
| :-: | :-: | :-: | :-: | :-: | :-: |
| Autotrophic | 3.43E-17| 3.41E-5| 4.51E-15 | --- | 1.52E-15 |
| Mixootrophic | 2.74E-17 |3.48E-5 | 1.05E-15 | 1.2E-15 | 3.4E-16 |
| Heterotrophic | 0.0 | 3.48E-5 | 0.0| 0.0 | --- |


Results show that biomass growth is decreased to a very low levels, what can be explained as a result of tyrosine production inhibition because tyrosine is considered by the model to calculate the BIOMASS flux. Regarding the p-coumaric production, it also decreases to values much more lower compared to wild-type. The study carried out by Gao et al. (2021) revealed an increase of 54% in p-coumaric production when tyrosine pathway was knocked out. Thus, our model don't predict the *in-vitro* results and further optimization regarding other knock out targets should be done. 

### 4.2: Additional biosynthetic pathway: Cloning of XAL enzyme

Another option to increase p-coumaric production would be to clone the gen encoding an enzyme able to directly convert L-tyrosine into p-coumaric acid in a sinlge reaction step. We select the enzyme XAL from *Trichosporon cutaneum*, which was already cloned and studied in *Synechocystis* sp. PCC6303 by Brel et al. (2020). XAL has both phenylalanine ammonia-lyase (PAL) and tyrosine ammonia-lyase (TAL) activity. Ideally, we clone the enzyme with a high constitutive expression system, which would redirect close to 100% tyrosine to p-coumaric conversion. Tyrosine and subproducts could be exogenously administered to ensure metabolic flux through the tyrosine related essential pathways. 

In [664]:
phenylalanine_ammonia_lyase2 = Reaction('PAL2')
tyrosine_ammonia_lyase = Reaction('TAL')
phenylalanine_ammonia_lyase2.add_metabolites(({model.metabolites.phe__L_c : -1,  
                              trans_Cinnamate: 1, 
                              model.metabolites.nh3_c: 1, 
                            }))
tyrosine_ammonia_lyase.add_metabolites(({model.metabolites.tyr__L_c : -1,                                                                                            
                                                  p_Coumaric_acid: 1,                                          
                                                  model.metabolites.nh3_c : 1,
                                                 }))

In [665]:
model.add_reactions([phenylalanine_ammonia_lyase2])
model.add_reactions([tyrosine_ammonia_lyase])


When we optimize the flux through any the three biomass reactions, we see that TAL flux is 0. The reason is that tyrosine is considered by the model as a metabolite taking part in biomass reactions. As the flux is being optimized through biomass, there is no flux left for TAL. Our solution is to consider an ideal situation where all the tyrosine is converted into p-coumaric, thus using tyrosine formation flux through TYRTA (Tyrosine transaminase) as the hypothetical TAL flux. 

In [666]:
with model as model:
    model.objective=model.reactions.BIOMASS_Ec_SynMixo_1
    maximum_growth_mixo = model.optimize().objective_value #flux
    print('Maximum growth =', maximum_growth_mixo, 'h-1')
    TAL_flux=model.reactions.TAL.flux
    print('TAL flux =', TAL_flux, 'mmol/g(DW)/h')
    TYRTA_flux=model.reactions.TYRTA.flux
    print('TYRTA flux =', -1*TYRTA_flux, 'mmol/g(DW)/h')

Maximum growth = 0.07802010063372718 h-1
TAL flux = 0.0 mmol/g(DW)/h
TYRTA flux = 0.01047497871108421 mmol/g(DW)/h


In [667]:
model.metabolites.tyr__L_c.reactions


frozenset({<Reaction BIOMASS_Ec_SynAuto_1 at 0x2a9f020ed00>,
           <Reaction BIOMASS_Ec_SynHetero_1 at 0x2a9f020eee0>,
           <Reaction BIOMASS_Ec_SynMixo_1 at 0x2a9f020e6d0>,
           <Reaction TAL at 0x2a9f0ab35b0>,
           <Reaction THZPSN2 at 0x2a9f06eab50>,
           <Reaction THZSN_1 at 0x2a9f0a2cf10>,
           <Reaction TYRTA at 0x2a9f07dec10>,
           <Reaction TYRTRS at 0x2a9f0663970>})

In [688]:
#Autotrophic
with model as model:
    model.reactions.EX_photon_e.bounds = -50, 0
    model.reactions.RBPC_1.bounds = 2.97, 3.54
    model.objective=model.reactions.BIOMASS_Ec_SynAuto_1
    maximum_growth_hetero = model.optimize().objective_value #flux
    print('Maximum growth =', maximum_growth_hetero, 'h-1')
    p_coumaric_production = -model.reactions.TYRTA.flux + model.reactions.CYP73A.flux
    print('Maximum productivity =', p_coumaric_production, 'mmol p-coumaric/gDW*h')
    maximum_yield_co2 = p_coumaric_production / (1*(model.reactions.RBPC_1.flux)) #yield calculations
    print('Maximum theoretical yield on co2 =', maximum_yield_co2, 'mmol-p_coumaric/mmol-co2')

Maximum growth = 0.17433033141836166 h-1
Maximum productivity = 0.023483458434071708 mmol p-coumaric/gDW*h
Maximum theoretical yield on co2 = 0.007906888361640305 mmol-p_coumaric/mmol-co2


In [669]:
#Mixotrophic
with model as model:
    model.reactions.EX_photon_e.bounds = 0, 95
    model.reactions.RBPC_1.bounds = 2.97, 3.54
    model.objective=model.reactions.BIOMASS_Ec_SynMixo_1
    maximum_growth_mixo = model.optimize().objective_value #flux
    print('Maximum growth =', maximum_growth_mixo, 'h-1')
    p_coumaric_production = -model.reactions.TYRTA.flux + model.reactions.CYP73A.flux
    print('Maximum productivity =', p_coumaric_production, 'mmol p-coumaric/gDW*h')
    maximum_yield_co2 = p_coumaric_production / (1*(model.reactions.RBPC_1.flux)) #yield calculations
    print('Maximum theoretical yield on co2 =', maximum_yield_co2, 'mmol-p_coumaric/mmol-co2')
    maximum_yield_glucose = p_coumaric_production / (-1*(model.reactions.EX_glc__D_e.flux)) #yield calculations
    print('Maximum theoretical yield on glucose =', maximum_yield_glucose, 'mmol-p-coumaric/mmol-glucose')

Maximum growth = 0.07802010063372702 h-1
Maximum productivity = 0.010509827949434065 mmol p-coumaric/gDW*h
Maximum theoretical yield on co2 = 0.003538662609237059 mmol-p_coumaric/mmol-co2
Maximum theoretical yield on glucose = 0.01236450346992243 mmol-p-coumaric/mmol-glucose


In [670]:
### Heterotrophic
with model as model:
    model.reactions.EX_photon_e.bounds = 0, 95
    model.reactions.RBPC_1.bounds = 0, 0
    model.objective=model.reactions.BIOMASS_Ec_SynHetero_1
    maximum_growth_hetero = model.optimize().objective_value #flux
    print('Maximum growth =', maximum_growth_hetero, 'h-1')
    p_coumaric_production = -model.reactions.TYRTA.flux + model.reactions.CYP73A.flux
    print('Maximum productivity =', p_coumaric_production, 'mmol p-coumaric/gDW*h')
    maximum_yield_glucose = p_coumaric_production / (-1*(model.reactions.EX_glc__D_e.flux)) #yield calculations
    print('Maximum theoretical yield on glucose =', maximum_yield_glucose, 'mmol-p-coumaric/mmol-glucose')

Maximum growth = 0.07802010063372711 h-1
Maximum productivity = 0.010509827949433717 mmol p-coumaric/gDW*h
Maximum theoretical yield on glucose = 0.01236450346992202 mmol-p-coumaric/mmol-glucose


##### Results

|  | Max. growth (h-1) | WT Max. productivity (mmol/g(DW)/h) | XAL Max. productivity (mmol/g(DW)/h)| Max. yield on glucose | Max yield on CO2
| :-: | :-: | :-: | :-: | :-: | :-: |
| Autotrophic | 0.174 | 7.78E-5| 0.0234 | --- | 0.0079 |
| Mixootrophic | 0.078 |3.48E-5 | 0.0105 | 0.012 | 0.0035 |
| Heterotrophic | 0.078 | 3.48E-5 | 0.0105 | 0.012 | --- |

These results show an increase in p-coumaric production with XAL. In the three scenarios, the increase in p-coumaric production is around 295 fold more. The study by Brel et al. (2020) showed an increase of 12 fold more, so our model predicts a higher improvement of p-coumaric production. However, further optimization of this cloning should be done, as we don't know if all the tyrosine can really be redirected through XAL flux *in vivo*. 

## References

>Brey, L. F., Włodarczyk, A. J., Thøfner, J. F. B., Burow, M., Crocoll, C., Nielsen, I., ... & Jensen, P. E. (2020). Metabolic engineering of Synechocystis sp. PCC 6803 for the production of aromatic amino acids and derived phenylpropanoids. Metabolic engineering, 57, 129-139.

>Brocks, J. J., Logan, G. A., Buick, R., & Summons, R. E. (1999). Archean molecular fossils and the early rise of eukaryotes. science, 285(5430), 1033-1036.

>Catherine, Q., Susanna, W., Isidora, E. S., Mark, H., Aurélie, V., & Jean-François, H. (2013). A review of current knowledge on toxic benthic freshwater cyanobacteria–ecology, toxin production and risk management. Water research, 47(15), 5464-5479.

>Du, W., Jongbloets, J. A., Hernández, H. P., Bruggeman, F. J., Hellingwerf, K. J., & dos Santos, F. B. (2016). Photonfluxostat: a method for light-limited batch cultivation of cyanobacteria at different, yet constant, growth rates. Algal research, 20, 118-125.

>Gademann K. (2011). Out in the green: biologically active metabolites produced by cyanobacteria. Chimia, 65(6), 416–419. https://doi.org/10.2533/chimia.2011.416

>Gao, E. B., Kyere-Yeboah, K., Wu, J., & Qiu, H. (2021). Photoautotrophic production of p-Coumaric acid using genetically engineered Synechocystis sp. Pasteur Culture Collection 6803. Algal Research, 54, 102180.

>Joshi, C. J., Peebles, C. A., & Prasad, A. (2017). Modeling and analysis of flux distribution and bioproduct formation in Synechocystis sp. PCC 6803 using a new genome-scale metabolic reconstruction. Algal research, 27, 295-310.

>Kaneko, T., Sato, S., Kotani, H., Tanaka, A., Asamizu, E., Nakamura, Y., ... & Tabata, S. (1996). Sequence analysis of the genome of the unicellular cyanobacterium Synechocystis sp. strain PCC6803. II. Sequence determination of the entire genome and assignment of potential protein-coding regions. DNA research, 3(3), 109-136.

>Montagud, A., Zelezniak, A., Navarro, E., de Córdoba, P. F., Urchueguía, J. F., & Patil, K. R. (2011). Flux coupling and transcriptional regulation within the metabolic network of the photosynthetic bacterium Synechocystis sp. PCC6803. Biotechnology journal, 6(3), 330-342.

>Nozzi, N. E., Oliver, J. W., & Atsumi, S. (2013). Cyanobacteria as a platform for biofuel production. Frontiers in bioengineering and biotechnology, 1, 7.

>Rasmussen, B., Fletcher, I. R., Brocks, J. J., & Kilburn, M. R. (2008). Reassessing the first appearance of eukaryotes and cyanobacteria. Nature, 455(7216), 1101-1104.

> Shastri, A. A., & Morgan, J. A. (2005). Flux balance analysis of photoautotrophic metabolism. Biotechnology progress, 21(6), 1617-1626.

>Touloupakis, E., Cicchi, B., & Torzillo, G. (2015). A bioenergetic assessment of photosynthetic growth of Synechocystis sp. PCC 6803 in continuous cultures. Biotechnology for biofuels, 8(1), 1-11.

>Van Alphen, P., Abedini Najafabadi, H., Branco dos Santos, F., & Hellingwerf, K. J. (2018). Increasing the photoautotrophic growth rate of Synechocystis sp. PCC 6803 by identifying the limitations of its cultivation. Biotechnology journal, 13(8), 1700764.

>Vermaas, W. (1996). Molecular genetics of the cyanobacteriumSynechocystis sp. PCC 6803: Principles and possible biotechnology applications. Journal of Applied Phycology, 8(4), 263-273.

>Vijayakumar, S., & Menakha, M. (2015). Pharmaceutical applications of cyanobacteria—A review. Journal of Acute Medicine, 5(1), 15-23.

>Yang, C., Hua, Q., & Shimizu, K. (2002). Metabolic flux analysis in Synechocystis using isotope distribution from 13C-labeled glucose. Metabolic engineering, 4(3), 202-216.

>Yu, Y., You, L., Liu, D., Hollinshead, W., Tang, Y. J., & Zhang, F. (2013). Development of Synechocystis sp. PCC 6803 as a phototrophic cell factory. Marine drugs, 11(8), 2894-2916.

>Zavřel, T., Očenášová, P., & Červený, J. (2017). Phenotypic characterization of Synechocystis sp. PCC 6803 substrains reveals differences in sensitivity to abiotic stress. PLoS One, 12(12), e0189130.
